src/I18n/RelativeTimeFormatter.php

```diff
@@ -87,15 +86,15 @@ public function diffForHumans(ChronosInterface $date, ?ChronosInterface $other =
     /**
      * Format a into a relative timestring.
      *
-     * @param \DateTimeInterface $time The time instance to format.
+     * @param \Cake\I18n\Time|\Cake\I18n\FrozenTime $time The time instance to format.
      * @param array $options Array of options.
      * @return string Relative time string.
      * @see \Cake\I18n\Time::timeAgoInWords()
      */
-    public function timeAgoInWords(DateTimeInterface $time, array $options = []): string
+    public function timeAgoInWords($time, array $options = []): string
     {
         $options = $this->_options($options, FrozenTime::class);
-        if ($options['timezone'] && $time instanceof ChronosInterface) {
+        if ($options['timezone']) {
```

Comment:
This looks like it will cause a regression, no?

Reply:
Not really. As per docblock this method only accepts `\Cake\I18n\Time` and `\Cake\I18n\FrozenTime` instances both of which have the `timezone()` method.

As @dereuromark mentioned above we ideally need a new interface which both `Time` and `FrozenTime` implement so that we can it as type hint and avoid any type checks inside methods.

Reply:
That said `RelativeTimeFormatter` isn't used directly by users so strong type hinting or defensive type checks aren't really necessary.